### PR TITLE
SingleDatePicker now re-evaluates isOutsideRange on componentWillReceiveProps

### LINKED ISSUE #549: isOutsideRange not re-evaluated in SingleDatePicker on render
Due to various factors, we are needing to use two SingleDatePickers instead of a DateRangePicker in our project (e.g., allowing open-ended selections). The problem we've run into is that when changing the start date in one picker, the second date picker won't re-evaluate `isOutsideRange` when gaining focus again, meaning that if the user moved the start date earlier they still wouldn't be able to pick an earlier end date until shifting between months and forcing a re-render.

Preparing a PR to address this.

### PR DESCRIPTION
**Pull request by jlyman**
Fixes #549 by adding a new call to `isOutsideRange` on `componentWillReceiveProps`, just like what is being done for `isDayBlocked` and `isDayHighlighted`.

### CODE REVIEW

File: src/components/SingleDatePicker.jsx [hunk 1]
```diff
@@ -218,6 +218,12 @@ export default class SingleDatePicker extends React.Component {
       values(visibleDays).forEach((days) => {
         Object.keys(days).forEach((day) => {
           const momentObj = moment(day);
+          if (isOutsideRange(momentObj)) {
```
Comment (timhwang21) [inline src/components/SingleDatePicker.jsx]: This might need to account for `this.isBlocked()` as well, maybe?

Comment (jlyman) [inline src/components/SingleDatePicker.jsx]: Could you elaborate for me @timhwang21? There are a couple other checks (`isDayBlocked()` and `isDayHighlighted()` that happen right afterwards, and I believe `this.isBlocked()` checks both `isDayBlocked` and `isOutsideRange` (on [line 628](https://github.com/jlyman/react-dates/blob/master/src/components/SingleDatePicker.jsx#L628)).

Comment (timhwang21) [inline src/components/SingleDatePicker.jsx]: i have to admit my memory is a bit fuzzy on this now but I believe isBlocked adds its own modifier on top of the others. So you can successfully remove the modifiers for isDayBlocked and isOutsideRange but still have the calendar day blocked. This is what i remember from my own pr, hopefully one of the maintainers can clarify.

Comment (majapw) [inline src/components/SingleDatePicker.jsx]: What @timhwang21 is saying is we should update this component to look more like what's in the `DayPickerRangeController`, ie:
```
if (didFocusChange) {
      values(visibleDays).forEach((days) => {
        Object.keys(days).forEach((day) => {
          const momentObj = moment(day);

          if (this.isBlocked(momentObj)) {
            modifiers = this.addModifier(modifiers, momentObj, 'blocked');
          } else {
            modifiers = this.deleteModifier(modifiers, momentObj, 'blocked');
          }

          if (isOutsideRange(momentObj)) {
            modifiers = this.addModifier(modifiers, momentObj, 'blocked-out-of-range');
          } else {
            modifiers = this.deleteModifier(modifiers, momentObj, 'blocked-out-of-range');
          }

          if (isDayBlocked(momentObj)) {
            modifiers = this.addModifier(modifiers, momentObj, 'blocked-calendar');
          } else {
            modifiers = this.deleteModifier(modifiers, momentObj, 'blocked-calendar');
          }

          if (isDayHighlighted(momentObj)) {
            modifiers = this.addModifier(modifiers, momentObj, 'highlighted-calendar');
          } else {
            modifiers = this.deleteModifier(modifiers, momentObj, 'highlighted-calendar');
          }
        });
      });
    }
```

**Comment by majapw** [comment]
This looks good! Just the one comment on adding the `this.isBlocked` => `--blocked` modifier as well.

File: src/components/SingleDatePicker.jsx (hunk 1 above)
Comment (jlyman) [inline src/components/SingleDatePicker.jsx]: Gotchya... Thanks @majapw and @timhwang21, that makes sense. Let me go ahead and make that improvement.

**Comment by vebjorni** [comment]
Is this feature in the v12.1.2?

**Comment by jlyman** [comment]
@vebjorni I don't believe so, as 12.1.2 was cut seven days ago, and this was merged 4 days ago.

**Comment by vebjorni** [comment]
@jlyman Thanks, that makes sense.

@majapw Any idea when the next version will be released?

**Pull request merged**